frontend/src/views/Game.tsx

```diff
@@ -41,8 +43,14 @@ function GamePage() {
   const [fullPageLoading, setFullPageLoading] = useState<boolean>(true);
   const [error, setError] = useState<string>('');
 
+  // When variable null, show nothing; otherwise, show notification.
+  const [gameNotification, setGameNotification] = useState<GameNotification | null>(null);
+
   // Variable to hold whether the user is subscribed to the primary Game socket.
-  const [socketSubscribed, setSocketSubscribed] = useState(false);
+  const [userSocketSubscribed, setUserSocketSubscribed] = useState(false);
+
+  // Variable to hold whether the user is subscribed to the notification socket.
+  const [notificationSocketSubscribed, setNotificationSocketSubscribed] = useState(false);
```

Comment:
Do you know how well these function when dealing with refreshes, CMD+SHIFT+T, back/forward buttons, etc? The main reason I ask is cause I know state can get cleared/act weird on refreshes and stuff, and I also noticed on a branch off the game timer stuff that refreshing the page led to [this](https://trello.com/c/w8Qjwlt4/220-refreshing-on-game-page-shows-a-socket-is-not-connected-error) happening, which idk if it's a false error, an actual connection issue, or an error with subscribing.

Reply:
These functions handle that _poorly_; the reason why that issue linked on Trello is happening is because there is no setup to **connect** to any socket on the Game page, as the code assumes the user is already connected. (Instead, we have the "Are you sure you want to leave?" check.)

This is definitely something that we'll fix in the future though (it may be the PR after my next one, but if you're working on it please go ahead) -- this change requires adding in the code to reconnect by default as long as the user is not navigating from the Game page. I may ask you how to handle CMD+SHIFT+T and back/forward buttons effectively when doing this 👍